File: src/main/java/com/skyspaceeleven/service/FlightCrewService.java
package com.skyspaceeleven.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspaceeleven.domain.FlightCrew;
import com.skyspaceeleven.dto.FlightCrewDTO;
import com.skyspaceeleven.dto.FlightCrewSearchDTO;
import com.skyspaceeleven.dto.FlightCrewPageDTO;
import com.skyspaceeleven.dto.FlightCrewConvertCriteriaDTO;
import com.skyspaceeleven.service.GenericService;
import com.skyspaceeleven.dto.common.RequestDTO;
import com.skyspaceeleven.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface FlightCrewService extends GenericService<FlightCrew, Integer> {

	List<FlightCrew> findAll();

	ResultDTO addFlightCrew(FlightCrewDTO flightCrewDTO, RequestDTO requestDTO);

	ResultDTO updateFlightCrew(FlightCrewDTO flightCrewDTO, RequestDTO requestDTO);

    Page<FlightCrew> getAllFlightCrews(Pageable pageable);

    Page<FlightCrew> getAllFlightCrews(Specification<FlightCrew> spec, Pageable pageable);

	ResponseEntity<FlightCrewPageDTO> getFlightCrews(FlightCrewSearchDTO flightCrewSearchDTO);
	
	List<FlightCrewDTO> convertFlightCrewsToFlightCrewDTOs(List<FlightCrew> flightCrews, FlightCrewConvertCriteriaDTO convertCriteria);

	FlightCrewDTO getFlightCrewDTOById(Integer flightCrewId);







}

File: skyspaceeleven_ui/src/services/FlightCrewService.js
import http from "../http-common"; 

class FlightCrewService {
  getAllFlightCrews(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/flightCrew/flightCrews`, searchDTO);
  }

  get(flightCrewId) {
    return this.getRequest(`/flightCrew/${flightCrewId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/flightCrew?field=${matchData}`, null);
  }

  addFlightCrew(data) {
    return http.post("/flightCrew/addFlightCrew", data);
  }

  update(data) {
  	return http.post("/flightCrew/updateFlightCrew", data);
  }
  
  uploadImage(data,flightCrewId) {
  	return http.postForm("/flightCrew/uploadImage/"+flightCrewId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new FlightCrewService();
